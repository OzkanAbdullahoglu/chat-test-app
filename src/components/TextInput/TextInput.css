.form-control-root {
    display: flex;
    height: 60px;
    box-sizing: border-box;
}

.base-input {
    font-family: 'BrownRegular';
    width: 100%;
    border: 0;
    height: 60px;
    display: flex;
    padding-left: 10px;
    box-sizing: content-box;
    background-color: #fffbf0;
    box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
}

.form-main {
    width: 100%;
    max-width: 480px;
    position: fixed;
    bottom: 0;
}

input::-webkit-contacts-auto-fill-button, 
input::-webkit-credentials-auto-fill-button {
  visibility: hidden;
  pointer-events: none;
  position: absolute;
  right: 0;
}