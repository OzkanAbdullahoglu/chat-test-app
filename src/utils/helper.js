/* eslint-disable linebreak-style */
import React, { useState, useEffect } from 'react';
export const timeConverter = (timeStamp) => {
  if (!timeConverter.cache) {
    timeConverter.cache = {};
  }

  if (!timeConverter.cache[timeStamp]) {
    const timeStampMs = timeStamp * 1000;
    const dateObj = new Date(timeStampMs);
    const dayOfMonth = dateObj.getDate() < 10 ? `0${dateObj.getDate()}` : dateObj.getDate();
    const months = [
      'January', 'February', 'March', 'April', 'May', 'June',
      'July', 'August', 'September', 'October', 'November', 'December',
    ];
    const curMonth = months[dateObj.getMonth()];
    const curYear = dateObj.getFullYear();
    const curMinute = dateObj.getMinutes() < 10 ? `0${dateObj.getMinutes()}` : dateObj.getMinutes();
    const curHour = (dateObj.getHours() < 10 ? `0${dateObj.getHours()}` : dateObj.getHours());

    timeConverter.cache[timeStamp] = {
      date: `${curMonth} ${dayOfMonth},${curYear}`,
      time: `${curHour}:${curMinute}`,
    };
    return timeConverter.cache[timeStamp];
  }
  return timeConverter.cache[timeStamp];
};

export const isToday = (timestamp, date) =>{
  const today = Math.floor(Date.now() / 1000).toString();
  console.log(today);
  return timeConverter(today).date === timeConverter(timestamp).date ?
    'Today' : timeConverter(today).date === timeConverter(timestamp).date - 1 ?
    'Yesterday' : date;
};




function getCurrentViewDims() {
  const { innerWidth: width, innerHeight: height } = window;
  return {
    width,
    height,
  };
}

export default function useCurrentViewDims() {
  const [windowDimensions, setWindowDimensions] = useState(getCurrentViewDims());

  useEffect(() => {
    function handleResize() {
      setWindowDimensions(getCurrentViewDims());
    }

    window.addEventListener('scroll', handleResize);
    return () => window.removeEventListener('scroll', handleResize);
  }, []);

  return windowDimensions;
}

