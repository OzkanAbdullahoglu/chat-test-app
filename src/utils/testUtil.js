/* eslint-disable linebreak-style */
import { createStore, applyMiddleware } from 'redux';
import { middlewares } from '../store';
import rootReducer from '../reducers';

const chat = {
  requestedData: [
    {
      id: 1,
      direction: 'in',
      status: 'read',
      timestamp: '1577834102',
      text: 'Alice was beginning',
    },
  ],
  isPending: false,
  error: '',
  scrollDownEnable: true,
};

const message = {
  messagesEndRef: '',
};

const version = { number: '0.0.1' };

export const initialStates = { chat, message, version };

export const storeFactory = (initalState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
  return createStoreWithMiddleware(rootReducer, initalState);
};
