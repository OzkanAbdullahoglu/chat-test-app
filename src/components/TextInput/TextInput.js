/* eslint-disable linebreak-style */
import React from 'react';
import { connect } from 'react-redux';
import { compose } from 'recompose';
import PropTypes from 'prop-types';

import { Wrapper } from '../Wrapper/Wrapper';
import { Icon } from '../Icon/Icon';
import './TextInput.css';
import {
  chatActions,
} from '../../reducers';

class TextInput extends React.Component {
  constructor(props) {
    super(props);
    this.state = { value: '' };
  }

    handleChange =(event) => {
      this.setState({ value: event.target.value });
    }

    handleSubmit = (event) => {
      const { setAddNewMessages, scrollToBottom } = this.props;
      setAddNewMessages(this.state.value);
      scrollToBottom();
      this.setState({ value: '' });
      event.preventDefault();
    }
    render() {
      return (
        <form className="form-main" onSubmit={this.handleSubmit} autoComplete={'off'}>
          <div className="form-control-root">
            <input
              placeholder="Sending Message..."
              id="standard-basic"
              type="text"
              className="base-input"
              onChange={this.handleChange}
              value={this.state.value}
            />
            <Wrapper
              sendIcon
            >
              <Icon
                sendIcon
                viewBox="0 0 24 24"
                onClick={this.handleSubmit}
              >
                <path d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z" />
                <path d="M0 0h24v24H0z" fill="none" />
              </Icon>
            </Wrapper>
          </div>
        </form>
      );
    }
}

TextInput.propTypes = {
  setAddNewMessages: PropTypes.func,
  scrollToBottom: PropTypes.func,
};

const mapStateToProps = (store) => ({});

const withRedux = connect(
  mapStateToProps,
  { ...chatActions }
);

export default compose(withRedux)(TextInput);
