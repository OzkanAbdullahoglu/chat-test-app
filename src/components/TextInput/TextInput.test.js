/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TextInput from './TextInput';
import { storeFactory } from '../../utils/testUtil';

describe('TextInput Component', () => {
  const store = storeFactory();
  const textInput = shallow(<TextInput store={store} />)
    .dive()
    .dive();
  it('renders without crashing', () => {
    expect(toJson(textInput)).toMatchSnapshot();
  });
});

describe('input changes on change', () => {
  it('it changes local state with on change event', () => {
    const store = storeFactory();
    const textInput = shallow(<TextInput store={store} />).dive().dive();
    textInput.find('input').simulate('change', { target: { value: 'test value' } });
    const inputValue = textInput.find('input').prop('value');
    expect(inputValue).toEqual('test value');
  });
});
