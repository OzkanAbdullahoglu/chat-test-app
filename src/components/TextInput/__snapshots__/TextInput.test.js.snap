// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component renders without crashing 1`] = `
<form
  autoComplete="off"
  className="form-main"
  onSubmit={[Function]}
>
  <div
    className="form-control-root"
  >
    <input
      className="base-input"
      id="standard-basic"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Sending Message..."
      type="text"
      value=""
    />
    <styled.div
      aria-label="Click to send a message"
      sendIcon={true}
      tabIndex="0"
    >
      <styled.svg
        data-test="svg-icon"
        onClick={[Function]}
        sendIcon={true}
        viewBox="0 0 24 24"
      >
        <path
          d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </styled.svg>
    </styled.div>
  </div>
</form>
`;
